packages/editor/src/components/provider/index.js

```diff
@@ -193,19 +194,26 @@ class EditorProvider extends Component {
 						type={ post.type }
 						id={ post.id }
 					>
-						<BlockEditorProvider
-							value={ blocks }
-							onInput={ resetEditorBlocksWithoutUndoLevel }
-							onChange={ resetEditorBlocks }
-							selectionStart={ selectionStart }
-							selectionEnd={ selectionEnd }
-							settings={ editorSettings }
-							useSubRegistry={ false }
+						<BlockContextProvider
+							value={ {
+								postId: post.id,
+								postType: post.type,
+							} }
 						>
-							{ children }
-							<ReusableBlocksButtons />
-							<ConvertToGroupButtons />
-						</BlockEditorProvider>
+							<BlockEditorProvider
+								value={ blocks }
+								onInput={ resetEditorBlocksWithoutUndoLevel }
+								onChange={ resetEditorBlocks }
+								selectionStart={ selectionStart }
+								selectionEnd={ selectionEnd }
+								settings={ editorSettings }
+								useSubRegistry={ false }
+							>
+								{ children }
+								<ReusableBlocksButtons />
+								<ConvertToGroupButtons />
+							</BlockEditorProvider>
+						</BlockContextProvider>
 					</EntityProvider>
```

Comment:
Do we still need the EntityProvider?

Reply:
> Do we still need the EntityProvider?

I'm still trying to piece together the full purpose of those entity contexts and to what extent these changes can supersede or deprecate those. I assume for backward-compatibility, we'll need to continue to support many of these (we even had [documented examples for migrating to `useEntityProp`](https://make.wordpress.org/core/2020/03/02/general-block-editor-api-updates/)). Would you think that we could stub those in some way where we could read instead from these "default" block context related to the current post?

Reply:
>  Would you think that we could stub those in some way where we could read instead from these "default" block context related to the current post?

Yes, this seems like a possible solution thought the package location for these things might be a blocker, Let's keep it and figure it out later.